apps/web-connect/src/features/checkout/components/PromoCodeRow.tsx

```diff
@@ -21,6 +21,17 @@ export function PromoCodeRow() {
 
     const [promo, setPromo] = useState(false);
 
+    //pull promo code from 
+    const queryString = window.location.search;
+    const queryParams = new URLSearchParams(queryString);
+    const prefilledPromoCode = queryParams.get("promoCode");
```

Comment:
@craigbranscom This is already done in the parent component `Checkout.tsx`. I think the only thing we might be missing is calling `handleApplyPromoCode();` in the Checkout.tsx if the `prefilledPromoCode` is defined. Let me know if that works.

Reply:
You're right, it needed the `handleApplyPromoCode()` but also had to fix a bug in `useEffect()` usage for that method to work properly.